crates/shared/src/bad_token/token_owner_finder/blockscout.rs

```diff
@@ -50,8 +60,20 @@ impl TokenOwnerFinding for BlockscoutTokenOwnerFinder {
             .append_pair("action", "getTokenHolders")
             .append_pair("contractaddress", &format!("{token:#x}"));
 
+        let metric = &Metrics::instance(crate::metrics::get_metric_storage_registry())
+            .unwrap()
+            .results;
         tracing::debug!("Querying Blockscout API: {}", url);
-        let response_text = self.client.get(url).send().await?.text().await?;
+        let response_text = match async { self.client.get(url).send().await?.text().await }.await {
```

Comment:
This might look funny. The reason for the async block is to chain the two await calls without making `?` return from the whole function before we can set metrics.

Reply:
Cute trick! I would have used `.and_then()` but I like this trick.

Reply:
Was considering that too. There is often a choice between the Future extension traits and an async block. Currently I slightly prefer the async block because it is integrated into the language.